airflow/providers/common/sql/hooks/sql.py

```diff
@@ -517,3 +518,30 @@ def test_connection(self):
             message = str(e)
 
         return status, message
+
+    def get_database_info(self, connection):
+        from airflow.providers.openlineage.utils.sqlparser import DatabaseInfo
+
+        return DatabaseInfo(
+            scheme=self.get_database_dialect(connection), authority=self.get_authority(connection)
+        )
+
+    def get_database_dialect(self, connection):
+        """Method used for SQL parsing. Naively tries to use Connection's conn_type"""
+        return connection.conn_type
+
+    def get_authority(self, connection):
```

Comment:
Is this envisaged/meant to be public or not?

```suggestion
    def _get_authority(self, connection):
```

Reply:
Done.